lib/config.dart

```diff
@@ -33,7 +33,7 @@ class Config {
       final breezSDK = injector.breezSDK;
       final breezConfig = await _getBundledConfig();
       final defaultConf = await _getDefaultConf(breezSDK, breezConfig.apiKey, breezConfig.nodeConfig);
-      final defaultMempoolUrl = defaultConf.mempoolspaceUrl;
+      final defaultMempoolUrl = defaultConf.mempoolspaceUrl ?? "https://mempool.space/";
```

Comment:
If the app explicitly sets it, only that instance will be used in the SDK, which will bypass the automatic fallback introduced in https://github.com/breez/breez-sdk/pull/898 .

I'd suggest to only set it if the user sets it.

Reply:
Also: the new base URL format is `https://mempool.space/api`.

Is that something the app can handle? (e.g. append `/api` and handle the trailing slashes?)

Reply:
> If the app explicitly sets it, only that instance will be used in the SDK, which will bypass the automatic fallback introduced in [breez/breez-sdk#898](https://github.com/breez/breez-sdk/pull/898) .
> 
> I'd suggest to only set it if the user sets it.

This `defaultMempoolUrl` is only used by application config and not passed sdk config.

> Also: the new base URL format is `https://mempool.space/api`.
> 
> Is that something the app can handle? (e.g. append `/api` and handle the trailing slashes?)

It's handled here: [BlockChainExplorerUtils](https://github.com/breez/c-breez/blob/main/lib/utils/blockchain_explorer_utils.dart) and tested.

Reply:
So if the user sets the mempool.space URL in the app settings, it won't be given as arg when initializing the SDK?

Reply:
> So if the user sets the mempool.space URL in the app settings, it won't be given as arg when initializing the SDK?

It will be passed to SDK only if explicitly set from app settings, if not SDK will treat it as empty and use the fallback mechanism.

Reply:
Asking because, with the change in https://github.com/breez/breez-sdk/pull/898 , it's now possible for the user to set their own mempool.space URL if so desired.

Reply:
Sounds good, thanks.

Reply:
> It will be passed to SDK only if explicitly set from app settings, if not SDK will treat it as empty and use the fallback mechanism.

Just as reminder, when and if it's set in the SDK, it has to be in the `https://mempool.space/api` format.